groups/views/groups.py

```diff
@@ -7,22 +9,39 @@
 class GroupList(ListView):
     """Show a top-level list of discussion groups."""
     model = models.Group
-    paginate_by = 5
     template_name = 'groups/group_list.html'
+    ordering = 'name'
 
 
 class GroupDetail(ListView):
     """Show the discussions belonging to a group."""
     model = models.Discussion
-    paginate_by = 5
+    paginate_by = 10
     template_name = 'groups/group_detail.html'
     subscribe_form_class = forms.SubscribeForm
 
+    def dispatch(self, request, *args, **kwargs):
+        """Get the group object we're accessing according to the pk in the URL."""
+        self.group = get_object_or_404(models.Group, pk=self.kwargs['pk'])
+        return super(GroupDetail, self).dispatch(request, *args, **kwargs)
+
     def get_queryset(self):
+        """Return all discussions on the particular group we're using."""
         return super(GroupDetail, self).get_queryset().filter(group=self.group)
 
+    def sort_object_list(self, object_list):
+        """
+        Order the list of discussions by time of most recent update.
+
+        Since this sorting irreversibly turns a queryset into a list, it's been separated
+        from get_queryset() to allow for extensibility on the latter.
+        """
+        return sorted(object_list, key=methodcaller('get_date_updated'), reverse=True)
```

Comment:
On a more serious note, if you have any tips on how to do that, I'd really appreciate them. :P I'm very concerned about the performance of that bit of functionality but didn't know how to work around that.  I haven't used annotation very much.